src/AtomicLaunch.sol

```diff
@@ -0,0 +1,281 @@
+    }
+
+    function _initialPoolPrice() internal returns (uint256 _poolPrice) {
+        (uint256 wbtcOraclePricing, , bool valid) = safeLatestAnswer(
+            wbtcUsdPriceFeed
+        );
+
+        require(valid, "not-valid!");
+
+        uint256 wbtcPrice = wbtcOraclePricing / 10**wbtcUsdPriceFeed.decimals();
+
+        _poolPrice = (wbtcPrice / CITADEL_PRICE) * 1e18;
+    }
+
+    function _deployOracles(address _ctdlWbtcCurvePool) internal {
+        CtdlWbtcCurveV2Provider ctdlWbtcProviderLoc = new CtdlWbtcCurveV2Provider(
+                _ctdlWbtcCurvePool
+            );
+        _oracles.add(address(ctdlWbtcProviderLoc));
+
+        CtdlBtcChainlinkProvider ctdlBtcProvider = new CtdlBtcChainlinkProvider(
+            _ctdlWbtcCurvePool,
+            WBTC_BTC_PRICE_FEED
+        );
+        _oracles.add(address(ctdlBtcProvider));
+
+        CtdlWibbtcLpVaultProvider ctdlWibbtcProvider = new CtdlWibbtcLpVaultProvider(
+                _ctdlWbtcCurvePool,
+                WBTC_BTC_PRICE_FEED,
+                WIBBTC_LP_VAULT
+            );
+        _oracles.add(address(ctdlWibbtcProvider));
+
+        CtdlEthChainlinkProvider ctdlEthProvider1 = new CtdlEthChainlinkProvider(
+                _ctdlWbtcCurvePool,
+                WBTC_BTC_PRICE_FEED,
+                BTC_ETH_PRICE_FEED
+            );
+        _oracles.add(address(ctdlEthProvider1));
+
+        CtdlAssetChainlinkProvider ctdlEthProvider2 = new CtdlAssetChainlinkProvider(
+                _ctdlWbtcCurvePool,
+                WBTC_BTC_PRICE_FEED,
+                address(wbtcUsdPriceFeed),
+                ETH_USD_PRICE_FEED
+            );
+        _oracles.add(address(ctdlEthProvider2));
+
+        address[4] memory assetEthFeeds = [
+            FRAX_ETH_PRICE_FEED,
+            USDC_ETH_PRICE_FEED,
+            CVX_ETH_PRICE_FEED,
+            BADGER_ETH_PRICE_FEED
+        ];
+
+        address[4] memory assetUsdFeeds = [
+            FRAX_USD_PRICE_FEED,
+            USDC_USD_PRICE_FEED,
+            CVX_USD_PRICE_FEED,
+            BADGER_USD_PRICE_FEED
+        ];
+
+        for (uint256 i; i < 4; ++i) {
+            CtdlAssetChainlinkProvider ctdlAssetProvider1 = new CtdlAssetChainlinkProvider(
+                    _ctdlWbtcCurvePool,
+                    WBTC_BTC_PRICE_FEED,
+                    BTC_ETH_PRICE_FEED,
+                    assetUsdFeeds[i]
+                );
+            _oracles.add(address(ctdlAssetProvider1));
+
+            CtdlAssetChainlinkProvider ctdlAssetProvider2 = new CtdlAssetChainlinkProvider(
+                    CTDL_WBTC_CURVE_POOL,
+                    WBTC_BTC_PRICE_FEED,
+                    address(wbtcUsdPriceFeed),
+                    assetEthFeeds[i]
+                );
+            _oracles.add(address(ctdlAssetProvider2));
+        }
+    }
+
+    function _fundingContractsInitialize() internal {
+        for (uint256 i = 0; i < _oracles.length(); i++) {
+            IFunding(_fundings.at(i)).initialize(
```

Comment:
IFunding interface is not imported. need to import it.

Reply:
should be resolved in latest commit